lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -0,0 +1,293 @@
+@pytest.fixture()
+def make_logger(server_config):
+    """
+    Construct a Pbench Logger object
+    """
+    return get_pbench_logger("TEST", server_config)
+
+
+class TestUnpackTarballs:
+    tarlist = ["/mock/ABC/TO-UNPACK/A.tar.xz", "/mock/ABC/TO-UNPACK/B.tar.xz"]
+    min_size, max_size, tar_size = 0, 1000, 586
+
+    def mock_stat(self, *args, follow_symlinks=False):
+        """Mocked replacement for Path.stat()"""
+        StatResult = namedtuple("StatResult", ["st_mode", "st_size"])
+        return StatResult(st_mode=16893, st_size=self.tar_size)
+
+    @pytest.mark.parametrize(
+        "min_size, max_size, total", [(586, 1000, 2), (0, 1000, 2), (0, 586, 0)]
+    )
+    def test_tarball_size(
+        self, monkeypatch, server_config, make_logger, caplog, min_size, max_size, total
+    ):
+        """Test for the valid size and path of the tarball"""
+
+        def mock_resolve(self: Path, strict: bool = False):
+            """
+            Mock replacement for Path.resolve()
+
+            Args:
+                strict: Boolean value to make sure path exist if True.
+
+            Returns:
+                Raising FileNotFoundError Exception
+            """
+            assert strict
+            raise FileNotFoundError(f"No such file or directory: '{self}'")
+
+        monkeypatch.setattr(Path, "glob", lambda path, args: self.tarlist)
+        monkeypatch.setattr(Path, "resolve", mock_resolve)
+        monkeypatch.setattr(Path, "stat", self.mock_stat)
+        obj = UnpackTarballs(server_config, make_logger)
+        result = obj.unpack_tarballs(min_size, max_size)
+        if result.total > 0:
+            for tar in self.tarlist:
+                assert (
+                    f"Tarball link, '{tar}', does not resolve to a real location: No such file or directory: '{tar}'"
+                    in caplog.text
+                )
+        assert result.total == total
+        assert result.success == 0
+
+    def test_unpack_md5_error(self, monkeypatch, server_config, make_logger, caplog):
+        """Show that when md5 file path cannot be accessed it throws a
+        FileNotFoundError Exception"""
+
+        def mock_md5sum(filepath: Path):
+            """
+            Mock of md5sum function
+
+            Args:
+                filepath: Tarball file path
+
+            Returns:
+                Raises FileNotFoundError Exception
+            """
+            raise FileNotFoundError(f"No such file or directory: '{filepath}.md5'")
+
+        monkeypatch.setattr("pbench.server.utils.md5sum", mock_md5sum)
+        file_tree = FileTree(server_config, make_logger)
+        obj = UnpackTarballs(server_config, make_logger)
+        for tar in self.tarlist:
+            with pytest.raises(Exception):
+                obj.unpack(Path(tar), file_tree)
+                assert (
+                    f"Getting md5 value of tarball '{tar}' failed: No such file or directory: '{tar}.md5'"
+                    in caplog.text
+                )
```

Comment:
we are calling obj.unpack for each tarball in tarlist.

Reply:
> we are calling obj.unpack for each tarball in tarlist.

Yes, but _why_ are we doing that?  😆  This test is exercising `UpackTarballs.unpack()` which operates on only one tarball.  So, there is no need to call it more than once...unless there is something about the subsequent calls which you think will provide a different behavior.  It could be that you think the tarball path is somehow significant, such that we need to try more than one of them; or, it could be that you think that state left over from a previous call will affect the behavior of a subsequent call; or, it could be something else that you're thinking of.

However, I don't see anything in the CUT source which suggests a need to call it more than once.  (So, I'm asking what you see....  🙂)

Reply:
No, there is nothing specific around that.